lib/remap-functions/src/is_nullish.rs

```diff
@@ -0,0 +1,154 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct IsNullish;
+
+impl Function for IsNullish {
+    fn identifier(&self) -> &'static str {
+        "is_nullish"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_) | Value::Null),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(IsNullishFn { value }))
+    }
+}
+
+#[derive(Clone, Debug)]
+struct IsNullishFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for IsNullishFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        match self.value.execute(state, object)? {
+            Value::Bytes(v) => {
+                let s = &String::from_utf8_lossy(&v)[..];
+
+                match s {
+                    "-" => Ok(true.into()),
+                    _ => {
+                        let has_whitespace = s.chars().all(char::is_whitespace);
+                        Ok(has_whitespace.into())
+                    }
+                }
+            }
+            Value::Null => Ok(true.into()),
+            _ => Err("input must be a string or null".into()),
```

Comment:
I would expect `is_nullish(["foo"])` to return `false`, not an error.

Maybe this shouldn't return an error, but return `false` in all other cases? That would make this function infallible as well, which is a nice property to strive for.

Reply:
I agree, I think ints, floats, arrays, etc. should just return false.

Reply:
Yeah, I agree. In the future we could potentially expand the definition of nullish to include 0 or `[]` or whatever we want based on user feedback (assuming that people even use this!) but for now this seems to do the job.